entity-management-definitions/src/main/java/eu/europeana/entitymanagement/normalization/EntityFieldsValidator.java

```diff
@@ -0,0 +1,299 @@
+				returnValue = checkURIFormat(context, field, fieldValueElem, returnValue, true);					
+			}
+			
+		}		
+        return returnValue;
+    }
+    
+    boolean validateEmailOrDateOrURIStringField(ConstraintValidatorContext context, Field field, Object fieldValue, boolean returnValue) {
+    	if (EntityFieldsTypes.valueOf(field.getName()).getFieldType().equals("Email")) {
+    		returnValue = checkEmailFormat(context, field, fieldValue, returnValue);
+    	}
+    	else if (EntityFieldsTypes.valueOf(field.getName()).getFieldType().equals("Date")) {
+    		returnValue = checkDateFormatISO8601(context, field, fieldValue, returnValue);
+    	}
+    	else if (EntityFieldsTypes.valueOf(field.getName()).getFieldType().equals("URI")) {
+			returnValue = checkURIFormat(context, field, fieldValue, returnValue, true);				
+		}
+        return returnValue;
+    }
+    
+    boolean checkEmailFormat(ConstraintValidatorContext context, Field field, Object fieldValue, boolean returnValue) {
+        String emailRegex = "^[a-zA-Z0-9_+&*-]+(?:\\."+
+                            "[a-zA-Z0-9_+&*-]+)*@" +
+                            "(?:[a-zA-Z0-9-]+\\.)+[a-z" +
+                            "A-Z]{2,7}$";                              
+        Pattern pat = Pattern.compile(emailRegex);
+        if (fieldValue == null) return false;
+        if (pat.matcher(fieldValue.toString()).matches()) {
+        	return true;
+        }
+        else
+        {
+    		addConstraint(returnValue, context, "The entity field: "+field.getName()+" is of type Email and contains inappropriate characters: "+fieldValue.toString()+".");
+    		return false;
+        }
+    }
+    
+    boolean checkDateFormatISO8601(ConstraintValidatorContext context, Field field, Object fieldValue, boolean returnValue) {
+    	try {
+        	//LocalDate.parse(fieldValue.toString(), DateTimeFormatter.ofPattern(java.time.format.DateTimeFormatter.ISO_DATE.toString()).withResolverStyle(ResolverStyle.STRICT));
+    		LocalDate.parse(fieldValue.toString(), java.time.format.DateTimeFormatter.ISO_DATE);
+        } catch (DateTimeParseException e) {
+    		addConstraint(returnValue, context, "The entity field: "+field.getName()+" is of type Date and does not comply with the ISO-8601 format: "+fieldValue.toString()+".");
+            returnValue = false;
+        }
+    	return returnValue;
+    }
+    
+    boolean checkURIFormat(ConstraintValidatorContext context, Field field, Object fieldValue, boolean returnValue, boolean shouldBeURI) {
+    	if (shouldBeURI && !EntityUtils.isUri(fieldValue.toString())) {
+			addConstraint(returnValue, context, "The entity field: "+field.getName()+" is of type: "+EntityFieldsTypes.valueOf(field.getName()).getFieldType()+" and it's element: "+fieldValue.toString()+" does not have the URI form.");
+			returnValue = false;
+		}
+    	else if (!shouldBeURI && EntityUtils.isUri(fieldValue.toString())) {
```

Comment:
this is not a requirement in the ticket EA-2420, please remove https://europeana.atlassian.net/browse/EA-2420

Reply:
I thought this is this requirement: Text: check if it is a string (not a reference). The same is for the Keyword.